src/planner/insert_plan.cpp

```diff
@@ -66,28 +70,46 @@ InsertPlan::InsertPlan(storage::DataTable *table,
   }
   // INSERT INTO table_name (col1, col2, ...) VALUES (val1, val2, ...);
   else {
-    PL_ASSERT(columns->size() <= schema->GetColumnCount());
+    uint32_t tup_size = ((*insert_values)[0]).size();  // size of each tuple
+    auto &table_columns = schema->GetColumns();
+    auto table_columns_num = schema->GetColumnCount();
+    // check for equality of columns size and insert values size
+    if (columns->size() > tup_size)
+      throw CatalogException(
+          "ERROR:  INSERT has more target columns than expressions");
+    else if (columns->size() < tup_size)
+      throw CatalogException(
+          "ERROR:  INSERT has more expressions than target columns");
+
+    std::vector<size_t> specified(
```

Comment:
Should I delete the `specified` vector defined here after the final loop where it is used? Or does C++ already do that, implicitly? Is there a potential memory leak here?